cmd/init.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2017 The Kedge Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"text/template"
+
+	"github.com/spf13/cobra"
+)
+
+type templateData struct {
+	Name  string
+	Image string
+	Ports []string
+}
+
+var (
+	fileName, image, name string
+	port                  []string
+)
+
+const (
+	boilerplate = `name: {{.Name}}{{"\n"}}containers:{{"\n"}}- image: {{.Image}}{{"\n"}}{{if .Ports}}services:{{"\n"}}- ports:{{block "list" .Ports}}{{"\n"}}{{range .}}{{println "  - port:" .}}{{end}}{{end}}{{end}}`
+)
+
+// initCmd represents the version command
+var initCmd = &cobra.Command{
+	Use:   "init",
+	Short: "Intialize kedge file",
+	Run: func(cmd *cobra.Command, args []string) {
+		data := templateData{Name: name, Image: image, Ports: port}
+		if name != "" && image != "" {
+			masterTmpl, err := template.New("master").Parse(boilerplate)
+			_, err = os.Stat(fileName)
+			if err != nil {
+				f, err := os.Create(fileName)
+				if err != nil {
+					fmt.Println(err, "failed to create file")
+					os.Exit(-1)
+				}
+				err = masterTmpl.Execute(f, data)
+				if err != nil {
+					fmt.Println(err, "failed to write file")
+					os.Exit(-1)
+				}
+				f.Close()
+				fmt.Println("file", fileName, "created")
+			} else {
+				fmt.Println(fileName, "is already present")
+				os.Exit(-1)
+			}
+		} else {
+			fmt.Println("--name and --image are mandatory flags, Please provide these flags")
+			os.Exit(-1)
+		}
+	},
+}
+
+func init() {
+	initCmd.Flags().StringVarP(&fileName, "file", "f", "kedge.yml", "Output filename")
```

Comment:
IMO `-f, --file` is confusing to use for specifying output filename since we are using it to accept filenames in other subcommands. How about `-o, --output`? ping @kadel